### PR TITLE
Add bindings for 'zmq_poll'.

### PR DESCRIPTION
**Pull request by civodul**
Hi again!

This adds bindings for `zmq_poll`.  This allows you to wait for read/write/error events from several sockets with code like:

```scheme
(zmq-poll (list (poll-item shell-socket) (poll-item iopub-socket)))
```

Ludo'.

PS: GitHub's "merge" button apparently creates gratuitous merge commits so  I'd recommend merging using the `git` command-line tool if possible.  :-)

### CODE REVIEW

**Comment by jerry40** [comment]
Hi! Excellent work, thanks! I think there is a high time for writing a test module, Can you give me a point how to do it correctly in a guile way? Also, let me press 'squash and merge' this time since you sent 1 commit. I will figure out the right set of git commands for merging PRs later.

File: src/simple-zmq.scm [hunk 1]
```diff
@@ -197,6 +211,23 @@
...
+  poll-item?
+  (socket   poll-item-socket)
+  (fd       poll-item-fd)
+  (events   poll-item-events))
+
+(define* (poll-item socket
+                    #:optional
+                    (events (logior ZMQ_POLLIN ZMQ_POLLOUT ZMQ_POLLERR))
+                    #:key (fd -1))
+  "Return a new \"poll item\" record suitable suitable for 'zmq-poll'.  The
```
Comment (jerry40) [inline src/simple-zmq.scm]: A typo "record suitable suitable for"

Comment (civodul) [reply src/simple-zmq.scm]: oops, indeed!

File: src/simple-zmq.scm [hunk 2]
```diff
@@ -504,3 +540,72 @@
...
+        (loop (+ index pollitem-size)
+              (match (parse-c-struct (pointer+ array index)
+                                     %pollitem-struct)
+                ((socket fd events 0)             ;no events
+                 result)
+                ((socket fd events revents)
+                 (cons (poll-item (if (null-pointer? socket)
+                                      #f
+                                      (pointer->socket socket))
+                                  events
```
Comment (civodul) [inline src/simple-zmq.scm]: Another typo: here it should be `revents`, not `events`.  Could you make this change?

**Comment by civodul** [comment]
With the fixes above it should be alright.  Let me know if you need anything else!

**Comment by civodul** [comment]
Regarding tests, I agree it's becoming seriously needed, though it shouldn't hold this PR back IMO.

The way I'd suggest doing it is to:

1. Properly package guile-simple-zmq, for instance using [Hall](https://gitlab.com/a-sassmannshausen/guile-hall), which can generate Autoconf/Automake machinery for you.
2. Add a `tests` directory containing test files that use the `(srfi srfi-64)` test suite library as many projects do.  Here's [an example](https://notabug.org/guile-sqlite3/guile-sqlite3/src/master/tests/basic.scm).
3. Add those tests to `Makefile.am` such that `make check` runs them, as in [this example](https://notabug.org/guile-sqlite3/guile-sqlite3/src/master/Makefile.am).

HTH!

**Comment by jerry40** [comment]
Thanks! This PR is ready to be merged, I'll do this in 3-4 hours (since I'm not ready to use cli to avoid squashing at the moment).

**Comment by jerry40** [comment]
I used cli commands,hopefully commits history is ok now. Thanks!

**Comment by civodul** [comment]
Great, thank you!

I personally like to rebase patches on top of the branch rather than
merge them, to keep the history linear as much as possible, but I
understand you might have different preferences.  :-)

**Comment by jerry40** [comment]
At  work we use rebase to master - squash - push force workflow, then one guy just presses 'merge' on bibucket side :) So I'm not very good at merging PRs. I need to read more about this process!

**Pull request merged**